_studio/shared/src/mfx_umc_mjpeg_vpp.cpp

```diff
@@ -267,10 +275,20 @@ mfxStatus mfx_UMC_FrameAllocator_D3D_Converter::StartPreparingToOutput(mfxFrameS
                 isD3DToSys = !((pOpaqAlloc->Out.Type & MFX_MEMTYPE_SYSTEM_MEMORY) == 0);
             }
 
-            pSrcTop.Info.CropW = surface_work->Info.CropW;
-            pSrcTop.Info.CropH = surface_work->Info.CropH / 2;
-            pSrcBottom.Info.CropW = surface_work->Info.CropW;
-            pSrcBottom.Info.CropH = surface_work->Info.CropH / 2;
+            if (par->mfx.Rotation == MFX_ROTATION_0 || par->mfx.Rotation == MFX_ROTATION_180)
+            {
+                pSrcTop.Info.CropW = surface_work->Info.CropW;
+                pSrcTop.Info.CropH = surface_work->Info.CropH / 2;
+                pSrcBottom.Info.CropW = surface_work->Info.CropW;
+                pSrcBottom.Info.CropH = surface_work->Info.CropH / 2;
+            }
+            else
+            {
+                pSrcTop.Info.CropW = surface_work->Info.CropH;
```

Comment:
I unable to test this functionality: for some reason vpp first runs field weaving to deinterlace, and does rotation on a separate step.